src/server/game.cpp

```diff
@@ -867,6 +867,15 @@ bool game::is_legal_command(const simple_wml::node& command, const socket_ptr us
 	if (!command.one_child()) return false;
 	// Chatting is never an illegal command.
 	if (command.child("speak")) return true;
+	if (command.child("surrender")) {
+		const simple_wml::string_span& sn = command.child("surrender")->attr("side_number");
+		if(sn.is_null() || sides_[sn.to_int()] != user) {
```

Comment:
Better change the key from side_number to from_side which already gets checked here. Otherwise you need to check that sn.to_int() is between 0 and sides_.size() - 1 before accessing sides_.